config/accounts.go

```diff
@@ -39,20 +44,147 @@ func (a *AccountCCPA) EnabledForIntegrationType(integrationType IntegrationType)
+	Purpose8                   AccountGDPRPurpose `mapstructure:"purpose8" json:"purpose8"`
+	Purpose9                   AccountGDPRPurpose `mapstructure:"purpose9" json:"purpose9"`
+	Purpose10                  AccountGDPRPurpose `mapstructure:"purpose10" json:"purpose10"`
+	// Hash table of purpose configs for convenient purpose config lookup
+	PurposeConfigs      map[consentconstants.Purpose]*AccountGDPRPurpose
+	PurposeOneTreatment AccountGDPRPurposeOneTreatment `mapstructure:"purpose_one_treatment" json:"purpose_one_treatment"`
+	SpecialFeature1     AccountGDPRSpecialFeature      `mapstructure:"special_feature1" json:"special_feature1"`
 }
 
 // EnabledForIntegrationType indicates whether GDPR is turned on at the account level for the specified integration type
-// by using the integration type setting if defined or the general GDPR setting if defined; otherwise it returns nil
+// by using the integration type setting if defined or the general GDPR setting if defined; otherwise it returns nil.
 func (a *AccountGDPR) EnabledForIntegrationType(integrationType IntegrationType) *bool {
-
 	if integrationEnabled := a.IntegrationEnabled.GetByIntegrationType(integrationType); integrationEnabled != nil {
 		return integrationEnabled
 	}
 	return a.Enabled
 }
+// PurposeEnforced checks if full enforcement is turned on for a given purpose at the account level. It returns the 
+// enforcement strategy type and whether or not it is set on the account. If not set, a default value of true is 
+// returned matching host default behavior.
+func (a *AccountGDPR) PurposeEnforced(purpose consentconstants.Purpose) (value, exists bool) {
+	if a.PurposeConfigs[purpose] == nil {
+		return true, false
+	}
+	if a.PurposeConfigs[purpose].EnforcePurpose == TCF2FullEnforcement {
+		return true, true
+	}
+	if a.PurposeConfigs[purpose].EnforcePurpose == TCF2NoEnforcement {
+		return false, true
+	}
+	return true, false
+}
+
+// PurposeEnforcingVendors gets the account level enforce vendors setting for a given purpose returning the value and
+// whether or not it is set. If not set, a default value of true is returned matching host default behavior.
+func (a *AccountGDPR) PurposeEnforcingVendors(purpose consentconstants.Purpose) (value, exists bool) {
+	if a.PurposeConfigs[purpose] == nil {
+		return true, false
+	}
+	if a.PurposeConfigs[purpose].EnforceVendors == nil {
+		return true, false
+	}
+	return *a.PurposeConfigs[purpose].EnforceVendors, true
+}
+
+// PurposeVendorException checks if the given bidder is a vendor exception for a given purpose.
+func (a *AccountGDPR) PurposeVendorException(purpose consentconstants.Purpose, bidder openrtb_ext.BidderName) (value, exists bool) {
+	if a.PurposeConfigs[purpose] == nil {
+		return false, false
+	}
+	if a.PurposeConfigs[purpose].VendorExceptionMap == nil {
+		return false, false
+	}
+	_, exists = a.PurposeConfigs[purpose].VendorExceptionMap[bidder]
+
+	return exists, true
```

Comment:
Is the order of `value` and `exist` in the function signature wrong? If so, can we correct? If not, can we rename `exists` to `value`?

Reply:
Another nitpick in regards to the consecutive `if` statements:
```
 99   // PurposeVendorException checks if the given bidder is a vendor exception for a given purpose.
100   func (a *AccountGDPR) PurposeVendorException(purpose consentconstants.Purpose, bidder openrtb_ext.BidderName) (value, exists bool) {
101 -     if a.PurposeConfigs[purpose] == nil {
102 -         return false, false
103 -     }
104 -     if a.PurposeConfigs[purpose].VendorExceptionMap == nil {
    +     if a.PurposeConfigs[purpose] == nil || a.PurposeConfigs[purpose].VendorExceptionMap == nil {
105           return false, false
106       }
107       _, exists = a.PurposeConfigs[purpose].VendorExceptionMap[bidder]
108
109       return exists, true
110   }
config/accounts.go
```

Reply:
> Is the order of `value` and `exist` in the function signature wrong? If so, can we correct? If not, can we rename `exists` to `value`?

The order is correct but I see how setting `exists` and then assigning it to `value` is confusing. How about changing `exists` to `found` when looking up `bidder` in the `VendorExceptionMap`:
```
_, found = a.PurposeConfigs[purpose].VendorExceptionMap[bidder]

return found, true
```
The second return value `exists` defined as a named return value represents whether `VendorExceptionMap` was defined. The `value` is whether it was found in the exception map.
to: